### PR TITLE
FIX Make CanView/Edit group toggle logic more robust

### PR DESCRIPTION
**Pull request by GuySartorelli**
Note that behat tests will be added to siteconfig as a part of https://github.com/silverstripe/silverstripe-siteconfig/issues/145

## Issue
- https://github.com/silverstripe/silverstripe-admin/issues/1609

### CODE REVIEW

File: client/src/legacy/LeftAndMain.EditForm.js [hunk 1]
```diff
@@ -569,7 +569,23 @@ $.entwine('ss', function($){
       }
     },
     onchange: function (e) {
-      this.toggleListDisplay(e.target.value);
+      this.toggleListDisplay(e.target.value, window.matchMedia('(prefers-reduced-motion: reduce)').matches);
```
Comment (ChloeSartorelli) [inline client/src/legacy/LeftAndMain.EditForm.js:572]: Technically out of scope but this just makes the animation not happen if the user as set their settings to prefer reduced motion. I can split this out if it's adding significant review overhead but I imagine it shouldn't do.

File: client/src/legacy/LeftAndMain.EditForm.js [hunk 2]
```diff
@@ -596,19 +612,20 @@ $.entwine('ss', function($){
     },
     showGroupsList: function (instant) {
       const holder = this.closest('.field');
-      this.showListElement(holder, holder.next().filter('.treemultiselect'), instant);
+      console.log(holder);
```
Comment (emteknetnz) [inline client/src/legacy/LeftAndMain.EditForm.js]: Need to remove the console.log statement

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: client/src/legacy/LeftAndMain.EditForm.js (hunk 2 above)
Comment (ChloeSartorelli) [reply client/src/legacy/LeftAndMain.EditForm.js]: Oops yup. Done.

**Review by emteknetnz — CHANGES_REQUESTED** [review]
The form field looks really janky when it init's when clicking the "Only these groups" radio.  It is existing behaviour because it does do this on the current non-animated buggy implementation on page load, though it feels significantly less bad because it just looks like a regular FOUT, rather than happening on click which feels much worse

https://www.youtube.com/watch?v=E3GJURQKyxU

Though the janky FOUT is existing behaviour, is there someway to get that to happen while it's sill invisible so that when the user clicks the radio is already been rendered correctly?

**Comment by GuySartorelli** [comment]
> is there someway to get that to happen while it's sill invisible so that when the user clicks the radio is already been rendered correctly?

Dunno. I guess we could put the initial hiding of the field in a delay, so it does the FOUT on page load and then hides the fields? I don't think that's really an improvement though. In any case I'd say that's out of scope probably. I'm not aware of any way to do it that doesn't require a worse on-page-load experience.

**Review by emteknetnz — APPROVED** [review]
Yeah probably out of scope since it's an existing issue.  Bugfix does what it's supposed to so will approve.

**Pull request merged**